src/web/components/Dropdown.tsx

```diff
@@ -157,132 +158,114 @@ const buttonExpanded = css`
-                }));
-                removeListeners();
-            };
-            const dismissOnEsc = (event: KeyboardEvent) => {
-                const escKey = 'Escape';
-
-                if (event.code === escKey) {
-                    this.setState(() => ({
-                        isExpanded: false,
-                    }));
-                    removeListeners();
-                }
-            };
-
-            document.addEventListener('keydown', dismissOnEsc, false);
-            document.addEventListener('click', dismissOnClick, false);
+        // Remove listener on unmount
+        return () => document.removeEventListener('click', dismissOnClick);
+    });
+
+    const dismissOnClick = (event: MouseEvent) => {
+        if (isExpanded) {
+            event.stopPropagation();
+            setIsExpanded(false);
         }
+    };
 
-        // needs to be unique to allow multiple dropdowns on same page
-        // this should be unique because JS is single-threaded
-        const dropdownID = `dropbox-id-${this.props.id}`;
-        const checkboxID = `checkbox-id-${this.props.id}`;
-
-        return (
-            <>
-                {this.state.noJS ? (
-                    <label
-                        htmlFor={checkboxID}
-                        className={cx({
-                            [button]: true,
-                            [buttonExpanded]: this.state.isExpanded,
-                        })}
-                        aria-controls={dropdownID}
-                        aria-expanded={this.state.isExpanded ? 'true' : 'false'}
-                        role="button"
-                    >
-                        <input
-                            className={input}
-                            type="checkbox"
-                            id={checkboxID}
-                            aria-controls={dropdownID}
-                            aria-checked="false"
-                            tabIndex={-1}
-                            key="OpenMainMenuCheckbox"
-                            role="menuitemcheckbox"
-                        />
-                        {label}
-                    </label>
-                ) : (
-                    <button
-                        onClick={this.boundToggle}
-                        className={cx({
-                            [button]: true,
-                            [buttonExpanded]: this.state.isExpanded,
-                        })}
+    const dismissOnEsc = (event: KeyboardEvent) => {
+        if (isExpanded && event.code === 'Escape') {
+            setIsExpanded(false);
+        }
+    };
+
+    const handleToggle = () => {
+        setIsExpanded(!isExpanded);
+    };
+
+    // needs to be unique to allow multiple dropdowns on same page
+    // this should be unique because JS is single-threaded
+    const dropdownID = `dropbox-id-${id}`;
+    const checkboxID = `checkbox-id-${id}`;
+
+    return (
+        <>
+            {noJS ? (
```

Comment:
Thoughts on refactoring out the two branches of the ternary to functions? Just because they're both long-ish and I think it would make it easier to mentally parse.

Reply:
I started to break this code out into separate components, a `DropdownLabel` and `DropdownLabelNoJS` but then I noticed the nojs version isn't working on DCR. So I'm going to create a new ticket to fix this (do we even need js for this component?) and leave this PR as is for now.